components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementFacade.java

```diff
@@ -331,6 +331,7 @@ private IdentityProvider populateEndpointConfig(IdentityProvider identityProvide
         }
         endpointConfigurationManager.resolveEndpointConfig(identityProvider.getFederatedAuthenticatorConfigs()[0],
                 tenantDomain);
+        identityProvider.setDefaultAuthenticatorConfig(identityProvider.getFederatedAuthenticatorConfigs()[0]);
```

Comment:
why this is needed?

Reply:
There was a null pointer exception when trying to get the default federated authenticator. Will reevalute this again and will send another PR, if they line is required.